xpdacq/calib.py

```diff
@@ -57,103 +52,215 @@ def _check_obj(required_obj_list):
-    """ function to run entire calibration process.
+    Entire process includes:
+    1) collect calibration image,
+    2) trigger pyFAI interactive calibration process,
+    3) store calibration parameters as a yaml file
 
-    Entire process includes: collect calibration image, trigger pyFAI
-    calibration process, store calibration parameters as a yaml file
-    under xpdUser/config_base/ and inject uid of calibration image to
-    following scans, until this function is run again.
+    calibration parameters will be saved under xpdUser/config_base/
+    and this set of parameters will be injected as metadata to
+    subsequent scans until you perform this process again
 
     Parameters
     ----------
     exposure : int, optional
-        total exposure time in sec. Default is 60s
+        total exposure time in sec. default is 5s
     dark_sub_bool : bool, optional
         option of turn on/off dark subtraction on this calibration
         image. default is True.
-    calibrant_file : str, optional
-        calibrant file being used, default is 'Ni.D' under
-        xpdUser/userAnalysis/. File name except for extention will be
-        used as sample name.
+   calibrant : str, optional
+        calibrant being used, default is 'Ni'.
+        input could be full file path to customized d-spacing file with
+        ".D" extension or one of pre-defined calibrant names.
+        List of pre-defined calibrant names is:
+        ['NaCl', 'AgBh', 'quartz', 'Si_SRM640', 'Ni', 'Si_SRM640d',
+         'Si_SRM640a', 'alpha_Al2O3', 'LaB6_SRM660b', 'TiO2', 'CrOx',
+         'LaB6_SRM660c', 'CeO2', 'Si_SRM640c', 'CuO', 'Si_SRM640e',
+         'PBBA', 'ZnO', 'Si', 'C14H30O', 'cristobaltite', 'LaB6_SRM660a',
+         'Au', 'Cr2O3', 'Si_SRM640b', 'LaB6', 'Al', 'mock']
     wavelength : float, optional
-        current of x-ray wavelength, in angstrom. Default value is
-        read out from existing xpdacq.Beamtime object
-    detector : pyfai.detector.Detector, optional.
-        instance of detector which defines pxiel size in x- and
-        y-direction. Default is set to Perkin Elmer detector
-    gaussian : int, optional
-        gaussian width between rings, Default is 100.
+        x-ray wavelength in angstrom. default to value stored in
+        existing Beamtime object
+    detector : str or pyFAI.detector.Detector instance, optional.
+        detector used to collect data. input could be instance of
+        Detector or one of pre-defined detector names. default to
+        'perkin_elmer' detector. please refer to pyFAI full
+        documentation for full list of pre-defined detectors
+    RE_instance : bluesky.run_engine.RunEngine instance, optional
+        instance of run engine being called in order to collect image.
+        DO NOT change it unless you are confident. default to current
+        run engine instance (xrun)
+    calib_collection_uid : str, optional
+        uid for this calibration run. default is to generate a new uid.
```

Comment:
uid for this calibration run.
By default a new uid is generated.  Do not change under normal circumstances.
override default when you want to associate this new calibration with an existing calibration-uid
in previously collected run headers.